File: src/main/java/fi/finlit/edith/sql/domain/Term.java
package fi.finlit.edith.sql.domain;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

@Entity
@Table(name = "term")
@Cache(usage=CacheConcurrencyStrategy.READ_WRITE)
public class Term {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String basicForm;

    private String meaning;

    @Enumerated(EnumType.STRING)
    private TermLanguage language;

    private String otherLanguage;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setMeaning(String meaning) {
        this.meaning = meaning;
    }

    public void setBasicForm(String basicForm) {
        this.basicForm = basicForm;
    }

    public String getMeaning() {
        return meaning;
    }

    public String getBasicForm() {
        return basicForm;
    }

    public void setOtherLanguage(String otherLanguage) {
        this.otherLanguage = otherLanguage;
    }

    public void setLanguage(TermLanguage language) {
        this.language = language;
    }

    public String getOtherLanguage() {
        return otherLanguage;
    }

    public TermLanguage getLanguage() {
        return language;
    }
}

File: src/main/scala/fi/finlit/edith/sql/domain/Term.scala
package fi.finlit.edith.sql.domain

import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import fi.finlit.edith.Identifiable
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

@Entity
@Table(name = "term")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
class Term extends Identifiable {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @BeanProperty
  var id: java.lang.Long = _

  @BeanProperty
  var basicForm: String = _

  @BeanProperty
  var meaning: String = _

  @Enumerated(EnumType.STRING)
  @BeanProperty
  var language: TermLanguage = _

  @BeanProperty
  var otherLanguage: String = _
}
